splash/qtrender_lua.py

```diff
@@ -415,16 +416,39 @@ def error(error_info):
         return cmd
 
     @command(async=True)
-    def go(self, url, baseurl=None, headers=None):
+    def go(self, url, baseurl=None, headers=None, http_method="GET", body=None, formdata=None):
+
         if url is None:
             raise ScriptError({
                 "argument": "url",
                 "message": "'url' is required for splash:go",
             })
 
+        http_method = http_method.upper()
+        if http_method not in ["POST", "GET"]:
+            raise ScriptError("Unsupported HTTP method {}".format(http_method))
+
+        if formdata and body:
+            raise ScriptError("formdata and body cannot be passed to go() in one call")
+
+        elif formdata:
+            body = self.lua.lua2python(formdata, max_depth=3)
+            if isinstance(body, dict):
+                body = urlencode(body)
+            else:
+                raise ScriptError("formdata argument for go() must be Lua table")
+
+        elif body:
+            body = self.lua.lua2python(body, max_depth=3)
```

Comment:
Yeah, but the only supported type is a string, and if I'm not mistaken lua2python is not needed to get a Python string from a Lua string because they are converted automatically. So you can check for a string and discard all other types without trying to convert them.